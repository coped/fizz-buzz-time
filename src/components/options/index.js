import Options from "components/options/Options";

export { Options };
