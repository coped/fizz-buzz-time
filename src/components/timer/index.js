import Timer from "components/timer/Timer"

export { Timer }