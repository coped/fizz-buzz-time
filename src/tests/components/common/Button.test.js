import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Button } from "components/common";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const text = "I'm a button";
  const props = {
    className: "test",
    onClick: () => console.log("Clicked"),
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("properly renders children", () => {
    const wrapper = mount(<Button>{text}</Button>);
    expect(wrapper.find("button").text()).toEqual(text);
  });

  it("properly handles props", () => {
    const wrapper = shallow(<Button {...props}></Button>);
    expect(wrapper.find("button").hasClass(props.className)).toEqual(true);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has a valid snapshot with props", () => {
    const component = renderer.create(<Button {...props}>{text}</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
