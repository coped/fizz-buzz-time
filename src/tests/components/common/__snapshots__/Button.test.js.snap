// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `<button />`;

exports[`Button has a valid snapshot with props 1`] = `
<button
  className="test"
  onClick={[Function]}
>
  I'm a button
</button>
`;
